horace_core/sqw/page_operations/PageOpBase.m

```diff
@@ -0,0 +1,204 @@
+classdef PageOpBase
+    %PAGEOPBASE class defines generic operation, performed on chunk of pixels
+    % located in memory.
+    %
+    % An operation normally consists of doing something with pixels and
+    % calculating appropriate avarages to image.
+    properties(Dependent)
+        % true if operation modifies PixelData only and does not affect
+        % image. The majority of operations modify both pixels and image
+        % Pixels only change in tests and in some pix-only operations
+        % e.g. recalc_pix_range
+        changes_pix_only;
+        % while page operations occur, not every page operation should be
+        % reported, as it could be too many logs. The property defines
+        % how many pages should be omitted until page progress is reported
+        log_split_ratio
+        % if provided, used as the name of the file for filebacked
+        % operations
+        outfile
+
+        signal_idx;
+        var_idx;
+        coord_idx;
+    end
+    properties(Dependent,Hidden)
+        % special property, containing pixels ordering when masking pixels
+        % only
+        npix
+    end
+
+
+    properties(Access=protected)
+        % holder for the pixel object, affected by the operation
+        pix_;
+        % holder for the image, modified by the operation.
+        img_;
+        % initial pixel range, recalculaed according to the operation
+        pix_data_range_ = PixelDataBase.EMPTY_RANGE;
+        %
+        outfile_   = '';
+        changes_pix_only_ = false;
+        log_split_ratio_  = 10;
+
+        signal_idx_;
+        var_idx_;
+        coord_idx_;
+
+        npix_ = [];
+    end
+    methods(Abstract)
+        % Specific apply operation method, which need overloading
+        % over
+        [obj,page_data] = apply_op(obj,npix_block,npix_idx,pix_id_first,pix_id_last);
+        %
+        % page_data -- block of pixel data modified by the operation
+    end
```

Comment:
If you look at the particular implementation(s), you will see the reason. It can be done through the function handle, but the need to reduce various sets of arguments and parameters makes the code huge mess. 

The subclassing allows you to focuse on operations and parameters, specific for the particular operation.  Function handle creates the terrible mess.